### PR TITLE
Rename callback blocks to completion to support Swift async and await

### LINKED ISSUE #91: Enable automatic async/await API generation
The GSI completion handlers use `callback` as the parameter selector for specifying completion handlers. 

`callback` is not one of the selectors included for automatic async method generation. See https://developer.apple.com/documentation/swift/calling_objective-c_apis_asynchronously

If I manually change the selector in the GIDSignIn.h public header to `completion`, Xcode will auto-generate the API:

Screen Shot 2022-01-14 at 2 54 16 PM

### PR DESCRIPTION
**Pull request by mdmathias**
Fixes #91

### CODE REVIEW

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDAuthentication.h [hunk 1]
```diff
@@ -64,9 +64,10 @@ typedef void (^GIDAuthenticationAction)(GIDAuthentication *_Nullable authenticat
 /// Get a valid access token and a valid ID token, refreshing them first if they have expired or are
 /// about to expire.
 ///
-/// @param action A callback block that takes a `GIDAuthentication` or an error if the attempt to
-///     refresh tokens was unsuccessful.  The block will be called asynchronously on the main queue.
-- (void)doWithFreshTokens:(GIDAuthenticationAction)action;
+/// @param completion A completion block that takes a `GIDAuthentication` or an error if the attempt
+///     to refresh tokens was unsuccessful.  The block will be called asynchronously on the main
+///     queue.
+- (void)doWithFreshTokens:(GIDAuthenticationAction)completion;
```
Comment (petea) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDAuthentication.h:70]: Does changing the variable name from `action` -> `completion` here trigger the async conversion?  My understanding is that conversion is selector driven.

Comment (mdmathias) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDAuthentication.h:70]: Yep, take a look at https://github.com/google/GoogleSignIn-iOS/pull/187/files#diff-47b5fba4ca251cba0d58e2e7bd57c2921e900575dca59d87414dc2db3b61b340R30

Comment (mdmathias) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDAuthentication.h:70]: Resolved as we are deferring testing this sort of thing to the sample app (or some API testing target to be defined in the future).

**Review by petea — COMMENTED** [review]
Did you intend to include related changes to DaysUntilBirthday here?

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDAuthentication.h (hunk 1 above)
Comment (petea) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDAuthentication.h:70]: Let's rename `GIDAuthenticationAction` to `GIDAuthenticationCompletion` for consistency.

**Comment by petea** [comment]
> Did you intend to include related changes to DaysUntilBirthday here?

n/m, I see #190 now.

**Review by petea — APPROVED** [review]

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDAuthentication.h (hunk 1 above)
Comment (Alex-4-Git) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDAuthentication.h:70]: This method will be moved to GIDGoogleUser. If we can agree on this change I will make the change in my PR.

File: GoogleSignIn/Sources/GIDSignIn.m [hunk 2]
```diff
@@ -217,52 +217,52 @@ - (BOOL)restorePreviousSignInNoRefresh {
 - (void)signInWithConfiguration:(GIDConfiguration *)configuration
        presentingViewController:(UIViewController *)presentingViewController
                            hint:(nullable NSString *)hint
-                       callback:(nullable GIDSignInCallback)callback {
+                       completion:(nullable GIDSignInCompletion)completion {
```
Comment (Alex-4-Git) [inline GoogleSignIn/Sources/GIDSignIn.m:220]: Align the colon.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDAuthentication.h (hunk 1 above)
Comment (mdmathias) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDAuthentication.h]: I'm adding it here. You can pull it it into your branch after I merge, which I'll do momentarily.

**Pull request merged**